crates/forge-runner/src/backtrace.rs

```diff
@@ -52,6 +61,52 @@ pub fn add_backtrace_footer(
         )
 }
 
+pub fn can_be_generated(scarb_metadata: &Metadata) -> Result<()> {
+    let manifest = fs::read_to_string(&scarb_metadata.runtime_manifest)?.parse::<DocumentMut>()?;
+
+    ensure!(
+        scarb_metadata.app_version_info.version >= MINIMAL_SCARB_VERSION,
+        "Coverage generation requires scarb version >= {MINIMAL_SCARB_VERSION}",
+    );
+
+    let has_needed_entries = manifest
+        .get("profile")
+        .and_then(|profile| profile.get(&scarb_metadata.current_profile))
+        .and_then(|profile| profile.get("cairo"))
+        .and_then(|cairo| cairo.as_table())
+        .is_some_and(|profile_cairo| {
+            CAIRO_BACKTRACE_REQUIRED_ENTRIES
+                .iter()
+                .all(|(key, value)| contains_entry_with_value(profile_cairo, key, value))
+        });
+
+    ensure!(
+        has_needed_entries,
+        formatdoc! {
+            "Scarb.toml must have the following Cairo compiler configuration to run backtrace:
+
+            [profile.{profile}.cairo]
+            unstable-add-statements-functions-debug-info = true
+            unstable-add-statements-code-locations-debug-info = true
+            inlining-strategy = \"avoid\"
+            ... other entries ...
+            ",
+            profile = scarb_metadata.current_profile
+        },
+    );
+
+    Ok(())
+}
+
+pub fn is_enabled() -> bool {
+    env::var(BACKTRACE_ENV).is_ok_and(|value| value == "1")
+}
+
+fn contains_entry_with_value(table: &Table, key: &str, value: &str) -> bool {
+    table
+        .get(key)
+        .is_some_and(|entry| entry.to_string().trim() == value)
```

Comment:
```suggestion
    table.get(key).is_some_and(|entry| {
        if let Some(entry) = entry.as_bool() {
            entry.to_string() == value
        } else {
            false
        }
    })
```
With the current implementation, it will fail if there is a comment. For your reference: https://github.com/foundry-rs/starknet-foundry/pull/2884